go-mode.el

```diff
@@ -1950,8 +1950,11 @@ If IGNORE-CASE is non-nil, the comparison is case-insensitive."
   (funcall go-packages-function))
 
 (defun go-packages-native ()
-  "Return a list of all installed Go packages.
-It looks for archive files in /pkg/."
+  "Return a list of all installed Go packages. Obsolete.
```

Comment:
Did you consider declaring this function as [obsolete](https://www.gnu.org/software/emacs/manual/html_node/elisp/Obsolete-Functions.html)?

Reply:
I did consider it. The reason I didn't do it was I didn't want to trigger the warning message (yet). Maybe I'm being too conservative but it seem like it would be fine to just switch the default and update the docs. I was thinking we would wait until a majority of projects have switched to modules and then use `(define-obsolete-function-alias)` to remove this function.